src/android/HappieCameraActivity.java

```diff
@@ -196,6 +196,15 @@ protected void initCameraSession() {
                     params.setFlashMode(Camera.Parameters.FLASH_MODE_TORCH);
                     flash.setImageResource(R.drawable.camera_flash_on);
             }
+            if(params.getSupportedPictureSizes()!=null) {
+                Camera.Size currentSize = params.getPictureSize();
+                Camera.Size maxSize = params.getSupportedPictureSizes().get(0);
+                if(currentSize.height<maxSize.height ||
+                        currentSize.width < maxSize.width) {
+                    params.setPictureSize(maxSize.width, maxSize.height);
+                }
+            }
+            params.setJpegQuality(100);
```

Comment:
Then we can keep 85%, I dont know that you did on purpose.

On Thursday 18 August 2016, James Spencer <notifications@github.com> wrote:

> In src/android/HappieCameraActivity.java
> <https://github.com/happieio/cordova-plugin-flex-camera/pull/5#discussion_r75223355>
> :
>
> > @@ -196,6 +196,15 @@ protected void initCameraSession() {
> >                      params.setFlashMode(Camera.Parameters.FLASH_MODE_TORCH);
> >                      flash.setImageResource(R.drawable.camera_flash_on);
> >              }
> > +            if(params.getSupportedPictureSizes()!=null) {
> > +                Camera.Size currentSize = params.getPictureSize();
> > +                Camera.Size maxSize = params.getSupportedPictureSizes().get(0);
> > +                if(currentSize.height<maxSize.height ||
> > +                        currentSize.width < maxSize.width) {
> > +                    params.setPictureSize(maxSize.width, maxSize.height);
> > +                }
> > +            }
> > +            params.setJpegQuality(100);
>
> @rishikapadia1 <https://github.com/rishikapadia1> thanks! is the
> resolution affected by setting setJpegQuality(85)? From from tests I did a
> while ago, 85 reduces the file size without noticeably reducing image
> quality.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/happieio/cordova-plugin-flex-camera/pull/5/files/9be28cedf96c08f075cfb4c1f508427a82606201#r75223355>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AErVjfKwktihlCW-3Qc5LbN6VUd6E9vWks5qg5LbgaJpZM4Jm8cr>
> .
>